lib/auth/accesspoint/accesspoint.go

```diff
@@ -83,9 +77,54 @@ func (c *AccessCacheConfig) CheckAndSetDefaults() error {
 	return nil
 }
 
-// NewAccessCache builds a cache.Cache instance for a teleport service. This logic has been
-// broken out of lib/service in order to support easier unit testing of process components.
-func NewAccessCache(cfg AccessCacheConfig) (*cache.Cache, error) {
+// NewAccessCacheForClient creates a new cache for a teleport service that
+// uses an authclient.ClientI to populate its resource collections.
+func NewAccessCacheForClient(cfg AccessCacheConfig, client authclient.ClientI) (*cache.Cache, error) {
```

Comment:
Can we finally rename this function to `NewCacheForClient`?

Reply:
5519c635a1